test/api/visualization/cluster_controller_test.rb

```diff
@@ -39,6 +39,10 @@ class ClusterControllerTest < ActionDispatch::IntegrationTest
   end
 
   teardown do
+    StudyFile.where(study_id: @basic_study.id).destroy_all
+    DataArray.where(study_id: @basic_study.id).destroy_all
```

Comment:
I don't understand why you can't trust the study destroy to delete these via cascade?  and if you can't, then lets move these subsidiary deletes to a helper method `delete_study_and_ensure_cascade` because otherwise it will be too easy to forget them in other test cleanups

Reply:
Deleting a study only deletes the `Study` object - there are no callbacks to remove other data.  We do this in normal operation through the `DeleteQueueJob` class since deletes take a while.  But I take the point - I do have to remember to add these cleanup tasks so that I can ensure the DB is clean between runs, so I'll refactor that into a helper method.

Reply:
0b7abb0